Makefile

```diff
@@ -26,7 +26,7 @@ node_modules/.bin/semver:
 test: lib
 	node scripts/ciTest.js -all
 
-test-syntax:
+test-syntax: build
```

Comment:
This will cause CI to run the dune build + copyExe.js twice.

I should probably clean this up somehow at some point, but for this PR I'd prefer to keep the Makefile as it is.

Reply:
Ah, noted. Reverted https://github.com/rescript-lang/rescript-compiler/pull/6238/commits/9af6b6a6cc43b97d356cf29aa685b3c70a3b88ed